Browser won't start in PROD mode

### Description

This is a regression caused by #6715.

The following exception is found in `swirlds.log` when starting a node with `netty.mode=PROD`:
```
2023-06-02 19:42:21.309 42       ERROR EXCEPTION        <main> Browser: exception on thread main
java.lang.UnsatisfiedLinkError: failed to load the required native library
        at io.netty.channel.epoll.Epoll.ensureAvailability(Epoll.java:80) ~[netty-transport-native-epoll-4.1.66.Final-linux-x86_64.jar:4.1.66.Final]
        at io.netty.channel.epoll.EpollEventLoop.<clinit>(EpollEventLoop.java:51) ~[netty-transport-native-epoll-4.1.66.Final-linux-x86_64.jar:4.1.66.Final]
        at io.netty.channel.epoll.EpollEventLoopGroup.newChild(EpollEventLoopGroup.java:185) ~[netty-transport-native-epoll-4.1.66.Final-linux-x86_64.jar:4.1.66.Final]
        at io.netty.channel.epoll.EpollEventLoopGroup.newChild(EpollEventLoopGroup.java:36) ~[netty-transport-native-epoll-4.1.66.Final-linux-x86_64.jar:4.1.66.Final]
        at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:84) ~[netty-common-4.1.77.Final.jar:4.1.77.Final]
        at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:60) ~[netty-common-4.1.77.Final.jar:4.1.77.Final]
        at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:49) ~[netty-common-4.1.77.Final.jar:4.1.77.Final]
        at io.netty.channel.MultithreadEventLoopGroup.<init>(MultithreadEventLoopGroup.java:59) ~[netty-transport-4.1.77.Final.jar:4.1.77.Final]
        at io.netty.channel.epoll.EpollEventLoopGroup.<init>(EpollEventLoopGroup.java:113) ~[netty-transport-native-epoll-4.1.66.Final-linux-x86_64.jar:4.1.66.Final]
        at io.netty.channel.epoll.EpollEventLoopGroup.<init>(EpollEventLoopGroup.java:100) ~[netty-transport-native-epoll-4.1.66.Final-linux-x86_64.jar:4.1.66.Final]
        at io.netty.channel.epoll.EpollEventLoopGroup.<init>(EpollEventLoopGroup.java:77) ~[netty-transport-native-epoll-4.1.66.Final-linux-x86_64.jar:4.1.66.Final]
        at io.netty.channel.epoll.EpollEventLoopGroup.<init>(EpollEventLoopGroup.java:53) ~[netty-transport-native-epoll-4.1.66.Final-linux-x86_64.jar:4.1.66.Final]
        at io.netty.channel.epoll.EpollEventLoopGroup.<init>(EpollEventLoopGroup.java:46) ~[netty-transport-native-epoll-4.1.66.Final-linux-x86_64.jar:4.1.66.Final]
        at com.hedera.node.app.service.mono.grpc.ConfigDrivenNettyFactory.configureProd(ConfigDrivenNettyFactory.java:88) ~[node-app-service-mono-0.40.0-SNAPSHOT.jar:?]
        at com.hedera.node.app.service.mono.grpc.ConfigDrivenNettyFactory.builderFor(ConfigDrivenNettyFactory.java:69) ~[node-app-service-mono-0.40.0-SNAPSHOT.jar:?]
        at com.hedera.node.app.service.mono.grpc.NettyGrpcServerManager.startOneNettyServer(NettyGrpcServerManager.java:84) ~[node-app-service-mono-0.40.0-SNAPSHOT.jar:?]
        at com.hedera.node.app.service.mono.grpc.NettyGrpcServerManager.start(NettyGrpcServerManager.java:69) ~[node-app-service-mono-0.40.0-SNAPSHOT.jar:?]
        at com.hedera.node.app.service.mono.grpc.GrpcStarter.startIfAppropriate(GrpcStarter.java:81) ~[node-app-service-mono-0.40.0-SNAPSHOT.jar:?]
        at com.hedera.node.app.MonoServicesMain.startNettyIfAppropriate(MonoServicesMain.java:119) ~[?:?]
        at com.hedera.node.app.MonoServicesMain.doStagedInit(MonoServicesMain.java:110) ~[?:?]
        at com.hedera.node.app.MonoServicesMain.initApp(MonoServicesMain.java:90) ~[?:?]
        at com.hedera.node.app.MonoServicesMain.init(MonoServicesMain.java:68) ~[?:?]
        at com.hedera.node.app.ServicesMain.init(ServicesMain.java:69) ~[?:?]
        at com.swirlds.platform.Browser.createLocalPlatforms(Browser.java:697) ~[swirlds-platform-core-0.39.0-adhoc.jar:?]
        at com.swirlds.platform.Browser.startPlatforms(Browser.java:830) ~[swirlds-platform-core-0.39.0-adhoc.jar:?]
        at com.swirlds.platform.Browser.<init>(Browser.java:312) ~[swirlds-platform-core-0.39.0-adhoc.jar:?]
        at com.swirlds.platform.Browser.launch(Browser.java:545) ~[swirlds-platform-core-0.39.0-adhoc.jar:?]
        at com.swirlds.platform.Browser.parseCommandLineArgsAndLaunch(Browser.java:496) ~[swirlds-platform-core-0.39.0-adhoc.jar:?]
        at com.swirlds.platform.Browser.main(Browser.java:854) ~[swirlds-platform-core-0.39.0-adhoc.jar:?]
Caused by: java.lang.NoSuchMethodError: Method 'void io.netty.channel.unix.Socket.initialize(boolean)' name or signature does not match
        at io.netty.channel.epoll.Native.registerUnix(Native Method) ~[netty-transport-native-epoll-4.1.66.Final-linux-x86_64.jar:4.1.66.Final]
        at io.netty.channel.epoll.Native.access$000(Native.java:52) ~[netty-transport-native-epoll-4.1.66.Final-linux-x86_64.jar:4.1.66.Final]
        at io.netty.channel.epoll.Native$1.run(Native.java:98) ~[netty-transport-native-epoll-4.1.66.Final-linux-x86_64.jar:4.1.66.Final]
        at io.netty.channel.unix.Unix.registerInternal(Unix.java:57) ~[netty-transport-native-unix-common-4.1.77.Final.jar:4.1.77.Final]
        at io.netty.channel.epoll.Native.<clinit>(Native.java:95) ~[netty-transport-native-epoll-4.1.66.Final-linux-x86_64.jar:4.1.66.Final]
        at io.netty.channel.epoll.Epoll.<clinit>(Epoll.java:39) ~[netty-transport-native-epoll-4.1.66.Final-linux-x86_64.jar:4.1.66.Final]
        at java.lang.Class.forName0(Native Method) ~[?:?]
        at java.lang.Class.forName(Unknown Source) ~[?:?]
        at io.grpc.netty.Utils.isEpollAvailable(Utils.java:299) ~[grpc-netty-1.48.2-module.jar:1.48.2]
        at io.grpc.netty.Utils.<clinit>(Utils.java:115) ~[grpc-netty-1.48.2-module.jar:1.48.2]
        at io.grpc.netty.NettyServerBuilder.<clinit>(NettyServerBuilder.java:84) ~[grpc-netty-1.48.2-module.jar:1.48.2]
        at com.hedera.node.app.service.mono.grpc.ConfigDrivenNettyFactory.builderFor(ConfigDrivenNettyFactory.java:67) ~[node-app-service-mono-0.40.0-SNAPSHOT.jar:?]
        ... 14 more
```
Note different `netty` versions in the stack trace.

### Steps to reproduce

Enable `netty.mode=PROD` (and `hedera.profiles.active=PROD`) in `hedera-node/data/config/node.properties`.
Start a node on Linux.

### Additional context

_No response_

### Hedera network

other

### Version

0.40.0-SNAPSHOT

### Operating system

Linux

--- a/buildSrc/src/main/kotlin/com.hedera.hashgraph.jpms-modules.gradle.kts
+++ b/buildSrc/src/main/kotlin/com.hedera.hashgraph.jpms-modules.gradle.kts
@@ -201,6 +201,10 @@ extraJavaModuleInfo {
         exportAllPackages()
         // no dependencies
     }
+    module("io.netty:netty-transport-native-epoll", "io.netty.transport.epoll") {
+        exportAllPackages()
+        // no dependencies
+    }
 
     knownModule("com.github.ben-manes.caffeine:caffeine", "com.github.benmanes.caffeine")
     knownModule("com.google.code.gson:gson", "com.google.gson")

--- a/buildSrc/src/main/kotlin/com/hedera/hashgraph/gradlebuild/rules/IoNettyNativeEpollMetadataRule.kt
+++ b/buildSrc/src/main/kotlin/com/hedera/hashgraph/gradlebuild/rules/IoNettyNativeEpollMetadataRule.kt
@@ -10,9 +10,10 @@ abstract class IoNettyNativeEpollMetadataRule : ComponentMetadataRule {
         val version = context.details.id.version
         context.details.allVariants {
             withFiles {
-                // Always pick 'linux-x86_64' by default
+                // Always pick 'linux-x86_64' and 'linux-aarch_64' by default
                 removeAllFiles()
                 addFile("$name-$version-linux-x86_64.jar")
+                addFile("$name-$version-linux-aarch_64.jar")
             }
         }
     }

--- a/hedera-node/hedera-mono-service/src/main/java/module-info.java
+++ b/hedera-node/hedera-mono-service/src/main/java/module-info.java
@@ -292,8 +292,8 @@ module com.hedera.node.app.service.mono {
     requires com.swirlds.logging;
     requires com.swirlds.platform;
     requires io.netty.handler;
-    requires io.netty.transport.epoll;
     requires io.netty.transport;
+    requires io.netty.transport.epoll;
     requires org.apache.commons.collections4;
     requires org.apache.commons.io;
     requires org.bouncycastle.provider;

--- a/hedera-node/test-clients/src/itest/resources/network/config/node.properties
+++ b/hedera-node/test-clients/src/itest/resources/network/config/node.properties
@@ -1,2 +1,5 @@
 hedera.recordStream.logDir=network/itest/data/recordStreams
-netty.mode=TEST
+# NK(2023-06-02): Integration tests should now run netty with the PROD configuration
+#                 which is necessary to close the coverage gap discovered by issue #6908.
+#                 (https://github.com/hashgraph/hedera-services/issues/6908)
+netty.mode=PROD

--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -134,7 +134,7 @@ dependencyResolutionManagement {
       val jacksonVersion = "2.13.3"
       val log4jVersion = "2.17.1"
       val mockitoVersion = "4.6.1"
-      val nettyVersion = "4.1.66.Final"
+      val nettyVersion = "4.1.77.Final"
       val swirldsVersion = "0.39.0-alpha.3"
       val systemStubsVersion = "2.0.2"
       val testContainersVersion = "1.17.2"

